csharp/NFT/NFTContract.cs

```diff
@@ -0,0 +1,234 @@
+        }
+
+        public static Iterator<byte[]> OwnerOf(ByteString tokenid)
+        {
+            return (Iterator<byte[]>)Storage.Find(Context, CreateStorageKey(Prefix_TokenOwner, tokenid));
+        }
+
+        public static Iterator<byte[]> TokensOf(UInt160 owner)
+        {
+            if (!owner.IsValid) throw new FormatException("The parameter 'owner' should be 20-byte address.");
+            return (Iterator<byte[]>)Storage.Find(Context, CreateStorageKey(Prefix_TokensOf, owner));
+        }
+
+        public static string Properties(ByteString tokenid)
+        {
+            byte[] key = CreateStorageKey(Prefix_Properties, tokenid);
+            return Storage.Get(Storage.CurrentContext, key);
+        }
+
+        public static bool Mint(ByteString tokenId, UInt160 owner, byte[] properties)
+        {
+            if (!Runtime.CheckWitness(superAdmin)) return false;
+
+            if (!owner.IsValid) throw new FormatException("The parameter 'owner' should be 20-byte address.");
+            if (properties.Length > 2048) throw new FormatException("The length of 'properties' should be less than 2048.");
+
+            StorageMap tokenOwnerMap = Storage.CurrentContext.CreateMap(CreateStorageKey(Prefix_TokenOwner, tokenId));
+            if (tokenOwnerMap.Get(owner) != null) return false;
+
+            StorageMap tokenOfMap = Storage.CurrentContext.CreateMap(CreateStorageKey(Prefix_TokensOf, owner));
+            byte[] key = CreateStorageKey(Prefix_Properties, tokenId);
+            Storage.Put(Storage.CurrentContext, key, (ByteString)properties);
+            tokenOwnerMap.Put(owner, owner);
+            tokenOfMap.Put(tokenId, tokenId);
+
+            var totalSupply = TotalSupply();
+            SetTotalSupply(totalSupply + FACTOR);
+
+
+            StorageMap tokenBalanceMap = Storage.CurrentContext.CreateMap(CreateStorageKey(Prefix_TokenBalance, owner));
+            tokenBalanceMap.Put(tokenId, FACTOR);
+
+            // Notify
+            MintTokenNotify(owner, tokenId, properties);
+            return true;
+        }
+
+        public static bool Burn(ByteString tokenId, UInt160 owner, BigInteger amount)
+        {
+            if (!owner.IsValid) throw new FormatException("The parameter 'owner' should be 20-byte address.");
+
+            if (amount < 0 || amount > FACTOR) throw new FormatException("The parameters 'amount' is out of range.");
+            if (amount == 0) return true;
+            if (!Runtime.CheckWitness(owner)) return false;
+
+            var tokenBalanceMap = Storage.CurrentContext.CreateMap(CreateStorageKey(Prefix_TokenBalance, owner));
+            var balanceValue = tokenBalanceMap.Get(tokenId);
+            var balance = balanceValue is null ? 0 : (BigInteger)balanceValue;
+            if (balance < amount) return false;
+
+            var totalSupply = TotalSupply();
+            balance -= amount;
+            totalSupply -= amount;
+            tokenBalanceMap.Put(tokenId, balance);
```

Comment:
Here you should also update the tokenOwnerMap and tokenOfMap when the balance == amount?